src/StorageRegistry.sol

```diff
@@ -0,0 +1,321 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.18;
```

Comment:
A few nits on contract versioning: 

1. Should we use standard versioning (v1, v2) or stick to the date versioning (2023.5.31) that we use for the protocol? Initial reaction is that having the date versions makes it easier to know which contract maps to which release, but I hold this weakly. 

2. Should the version number go into the name of the contract? StorageRegistryv1 instead of StorageRegistry

Reply:
If the contracts are coupled to a protocol version, I think the same format makes sense for both. I'd prefer adding a `VERSION` constant to each contract rather than using the contract name (especially if we're using the date-based protocol version format).